src/main/java/lotto/application/LottoService.java

```diff
@@ -0,0 +1,57 @@
+package lotto.application;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import lotto.domain.LottoLotteries;
+import lotto.domain.LottoLottery;
+import lotto.domain.LottoNumber;
+import lotto.domain.LottoResults;
+import lotto.domain.Message;
+import lotto.domain.Money;
+import lotto.domain.WinLottoNumbers;
+import lotto.utils.RandomLottoNumberGenerator;
+
+public class LottoService {
+	private final LottoLotteries lottoLotteries;
+
+	public LottoService(Money money) {
+		this.lottoLotteries = buyMaxLottoLotteries(money);
+	}
+
+	public LottoLotteries buyMaxLottoLotteries(Money money) {
+		int count = money.buyMax(LottoLottery.LOTTO_PRICE_PER_PIECE);
+		validateBuyLottoLottery(count);
+
+		return buyAutoLottoLotteries(count);
+	}
+
+	private void validateBuyLottoLottery(int count) {
+		if (count == 0) {
+			throw new IllegalArgumentException(Message.INVALID_BUY_LOTTO);
+		}
+	}
+
+	private LottoLotteries buyAutoLottoLotteries(int count) {
+		return new LottoLotteries(createAutoLottoLotteries(count));
```

Comment:
일급컬렉션에서 직접 객체 생성하도록 만들면 어떨까요?

```suggestion
		return new LottoLotteries(count, randomLottoNumberGenerator);
```

Reply:
일급컬렉션에 생성을 위임하는게 더 맞는것 같습니다!